src/CoreWCF.RabbitMQ.Client/src/CoreWCF/ServiceModel/Channels/RabbitMqOutputChannel.cs

```diff
@@ -0,0 +1,133 @@
+
+        public override T GetProperty<T>()
+        {
+            if (typeof(T) == typeof(System.ServiceModel.Channels.IOutputChannel))
+            {
+                return (T)(object)this;
+            }
+
+            T messageEncoderProperty = _encoder.GetProperty<T>();
+            if (messageEncoderProperty != null)
+            {
+                return messageEncoderProperty;
+            }
+
+            return base.GetProperty<T>();
+        }
+
+        /// <summary>
+        /// Open the channel for use. We do not have any blocking work to perform so this is a no-op
+        /// </summary>
+        protected override void OnOpen(TimeSpan timeout)
+        { }
+
+        protected override IAsyncResult OnBeginOpen(TimeSpan timeout, AsyncCallback callback, object state)
+        {
+            return Task.CompletedTask;
+        }
+
+        protected override void OnEndOpen(IAsyncResult result)
+        { }
+
+        protected override void OnAbort()
+        { }
+
+        protected override void OnClose(TimeSpan timeout)
+        {
+            _rabbitMqClient.Close();
+        }
+
+        protected override IAsyncResult OnBeginClose(TimeSpan timeout, AsyncCallback callback, object state)
+        { return Task.CompletedTask; }
+
+        protected override void OnEndClose(IAsyncResult result)
+        { }
+
+        /// <summary>
+        /// Address the Message and serialize it into a byte array.
+        /// </summary>
+        private ArraySegment<byte> EncodeMessage(System.ServiceModel.Channels.Message message)
+        {
+            try
+            {
+                _baseAddress.ApplyTo(message);
+                return _encoder.WriteMessage(
+                    message,
+                    (int)_parent.Transport.MaxReceivedMessageSize,
+                    _parent.BufferManager);
+            }
+            finally
+            {
+                // We have consumed the message by serializing it, so clean up
+                message.Close();
+            }
+        }
+
+        public void Send(System.ServiceModel.Channels.Message message)
+        {
+            var messageBuffer = EncodeMessage(message);
+
+            try
+            {
+                _rabbitMqClient.BasicPublish(
+                    exchange: _connectionSettings.Exchange,
+                    routingKey: _connectionSettings.RoutingKey,
+                    body: messageBuffer);
+            }
+            finally
+            {
+                // Make sure buffers are always returned to the BufferManager
+                _parent.BufferManager.ReturnBuffer(messageBuffer.Array);
+            }
+        }
+
+        public void Send(System.ServiceModel.Channels.Message message, TimeSpan timeout)
+        {
+            Send(message);
+        }
+
+        public IAsyncResult BeginSend(System.ServiceModel.Channels.Message message, AsyncCallback callback, object state)
+        { return Task.CompletedTask; }
```

Comment:
BeginSend is what gets called if your contract is async (eg returns `Task<Foo>`) and this is a no-op. Make a private SendAsync and call that from BeginSend/EndSend using ToApm and ToApmEnd.